### ISSUE TITLE
Container element constraints not recognized in generated openapi.json

### ISSUE THREAD

**Issue by varinderkaur61** [swagger-api/swagger-core#4224]
With hibernate 6.x+ versions, we have the support for container element constraints i.e.
`List<@Pattern(regexp="<sample pattern>") String> listElement;`

This is to validate each item under the list with the regex provided inside diamond operator. It do validates fine but the open api.json doesn't recognize this annotation and in resulting open api specs, it doesn't show any pattern or validation for individual items.

Expectation is it should show the applied pattern under items of list/array i.e.: 

The generated open api.json for the above schema object:
```
"components": {
        "schemas": {
            "listElement": {
                "type": "array",
                "items": {
                    "type": "string"
                }
            }
        }
  }
```
The expected/required open api specs:
```
"components": {
        "schemas": {
            "listElement": {
                "type": "array",
                "items": {
                    "type": "string",
                     "pattern": "<sample pattern>"
                }
            }
        }
  }
```

**Comment by NovelleCode** [comment]
Running into the same problem, would be great to have a solution for this!

**Comment by frantuma** [comment]
fixed by #4373
